### PR TITLE
feat(sdk): rfc for `cloud.Stream` and AWS support

### PR DESCRIPTION
**Pull request by 5herlocked**
Implementing design and library spec supporting and integrating real-time streaming services.

Streaming data services allow real-time processing of continuously generated high-velocity data to enable instant analytics and insights. They provide dynamic scaling, fault tolerance and distributed processing make them a natural fit for building scalable, reliable and low-latency systems in the cloud.

Supporting #1333

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/cloud/stream.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+## Advanced Usage
+
+If the expected throughput of your data is static and known, you can leverage provisioned throughput.
+
+```ts
+bring cloud;
+
+let stream = new cloud.Stream(
+  name: ,
```
Comment (Chriscbr) [inline libs/wingsdk/src/cloud/stream.md:43]: The name can probably be omitted -- each preflight class has a construct ID that can be used to customize for the resource's name. The syntax might change in the future but right now it's customizable like this:

```ts
let stream = new cloud.Stream(
   ...options
) as "MyStream";
```

File: libs/wingsdk/src/cloud/stream.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+if let lambdaFn = aws.Function.from(f) {
+  lambdaFn.addPolicyStatements(
+    aws.PolicyStatement {
+      actions: ["ses:sendEmail"],
+      effect: aws.Effect.ALLOW,
+      resources: ["*"],
+    },
+  );
+}
+```
```
[the remark this answers is not in the document]
Comment (5herlocked) [reply libs/wingsdk/src/cloud/stream.md:79]: hold over from copying a different segment of documentation. Should've cleaned it up.

File: libs/wingsdk/src/cloud/stream.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Most cloud providers have a data horizon for their streaming data to help providers reduce cost, prevent unbounded data growth, encourage real-time processing, and implement data governance best practices. However, developers have the ability to modify the data horizon.
+
+## Usage
+
+```ts
+bring cloud;
+
+let stream = new cloud.Stream(
+    name: 'test-data-stream',
+    horizon: 96,  // optional, defaults to 24 hours
```
[the remark this answers is not in the document]
Comment (5herlocked) [reply libs/wingsdk/src/cloud/stream.md:31]: Ack

**Comment by eladb** [comment]
It's very exciting to see this!

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+let stream = new cloud.Stream(
+    name: 'telemetry-ingest',
+    horizon: 48
+);
+
+let bloc = new cloud.Function(inflight (event: any): any => {
+    let newestBatch = stream.fetch();
+
+    let batchSize = newestBatch.size();
```
[the remark this answers is not in the document]
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md:69]: Any recommendations on how to codify it?

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+* IoT - Collecting and processing telemetry streams from IoT devices.
+* Gaming - Processing real-time game data streams for features like leaderboards or in-game alerts.
+* Financial Services - Performing real-time analytics and complex event processing on financial data feeds.
+* E-Commerce - Real-time monitoring of buying behaviour, stock levels, logistics data to enable instant actions.
+
+This is implemented across varied cloud providers, specifically services like: AWS Kinesis Data Streams, Google Cloud Pub/Sub, Azure Event Hubs and in open source solutions like Apache Kafka, Apache Pulsar, etc.
+
+## Design
+
+Traditionally, a device/user writing to a real-time streaming system would simply write to the REST API endpoint exposed by the service. While consumers of the service have to
```
Comment (eladb) [inline docs/contributing/999-rfcs/1333-cloud-stream-spec.md:35]: unfinished sent...

;-)

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+Within wing, these endpoints and their nuance should be abstracted, so writing to or reading from a streaming system is just like writing to or reading from any persistent storage.
+
+For example:
+
+Writing to a data stream from a function should look like:
+```ts
+bring cloud;
+
+let stream = new cloud.Stream(
+    horizon: 48h
```
Comment (eladb) [inline docs/contributing/999-rfcs/1333-cloud-stream-spec.md:46]: nit: we use 2 space indent

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+) as "TelemetryIngest";
+
+
+let bloc = new cloud.Function(inflight (event: any): any => {
+    stream.send(event);
+
+    let eventSize = event.count;
+    return argsCount % 2;
```
Comment (eladb) [inline docs/contributing/999-rfcs/1333-cloud-stream-spec.md:54]: what's the return value here?

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+```
+
+## Requirements
+
+### Functional
+
+- Stream-01 (P0): Create a Streaming Data Endpoint in the cloud of your choice
+- Stream-02 (P1): Read, write APIs to transparently expose the underlying endpoint
+- Stream-06 (P2): Implicit integration with Stream Aggregration services
```
Comment (eladb) [inline docs/contributing/999-rfcs/1333-cloud-stream-spec.md:104]: what about sharding?

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md (hunk 7 above)
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md:54]: Good point, cleaning it up.

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md (hunk 6 above)
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md:46]: ack

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md (hunk 8 above)
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md:104]: ack, Stream-07 (P2)

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md (hunk 5 above)
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md:35]: finished sentence ;D

File: docs/contributing/999-rfcs/1333-cloud-stream-spec.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (revitalbarletz) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md]: @Chriscbr any recommendations for @5herlocked ?

[the remark this answers is not in the document]
Comment (5herlocked) [reply docs/contributing/999-rfcs/1333-cloud-stream-spec.md]: I'm actually past that question. Ty for the check-in 😅

File: apps/vscode-wing/src/console/console-manager.ts [hunk 9]
```diff
@@ -230,6 +237,7 @@ export const createConsoleManager = (
       webviewPanel.onDidDispose(async () => {
         resourcesExplorer.clear();
         testsExplorer.clear();
+        endpointsExplorer.clear();
```
Comment (eladb) [inline apps/vscode-wing/src/console/console-manager.ts:240]: rebase from `main`

Comment (5herlocked) [reply apps/vscode-wing/src/console/console-manager.ts:240]: Will make the changes asap, sorry forgot what merge ff does to the git history

**Comment by garysassano** [comment]
I believe [WarpStream](https://www.warpstream.com/) could be a nice way to implement this.

**Comment by staycoolcall911** [comment]
@5herlocked - I'll go ahead and close this PR for housekeeping. If you're interested to pick it up again we'll be very happy to help! Let us know here or on our Discord: https://t.winglang.io/discord

**Pull request closed without merging**